### PR TITLE
fix(route): xiaomiyoupin crowdfunding error and migrate to v2

### LINKED ISSUE #11640: 小米有品众筹 出错
### Routes

```routes
/mi/youpin/crowdfunding
```

### Full routes

```fullroutes
/mi/youpin/crowdfunding
```

### Related documentation

https://docs.rsshub.app/shopping.html#xiao-mi

### What is expected?

拉取小米有品众筹

### What is actually happening?

this route is empty, please check the original site or [create an issue](https://github.com/DIYgod/RSSHub/issues/new/choose)

### Deployment information

RSSHub demo (https://rsshub.app)

### Deployment information (for self-hosted)

_No response_

### Additional info

```shell
Route requested: /mi/youpin/crowdfunding
Error message: this route is empty, please check the original site or create an issue
Helpful Information to provide when opening issue:
Path: /mi/youpin/crowdfunding
Node version: v16.19.0
Git Hash: 5456d13
```

### This is not a duplicated issue

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) to ensure this bug has not already been reported

### PR DESCRIPTION
**Pull request by bigfei**
## 该 PR 相关 Issue / Involved issue

Close #11640

## 完整路由地址 / Example for the proposed route(s)

```routes
/xiaomiyoupin/crowdfunding
/xiaomiyoupin/latest
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
1. Use the new API to fetch the crowdfunding items.
2. migrate to v2 route standard.

### CODE REVIEW

File: docs/shopping.md [hunk 1]
```diff
@@ -461,19 +461,15 @@ For instance, in <https://www.leboncoin.fr/recherche/?**category=10&locations=Pa
...
 ### 小米有品每日上新
 
-<Route author="xyqfer DIYgod" example="/mi/youpin/new" path="/mi/youpin/new/:sort?" :paramsDesc="['排序，见下表']">
-
-| 个性化排序 | 按销量从高到低 | 按好评从高到低 | 按上新时间从近到远 |
-| ----- | ------- | ------- | --------- |
-| 0     | 1       | 2       | 3         |
-
-</Route>
+<Route author="bigfei" example="/xiaomiyoupin/latest" path="/xiaomiyoupin/latest" />
```
Comment (TonyRL) [inline docs/shopping.md:472]:
```suggestion
<Route author="xyqfer DIYgod bigfei" example="/xiaomiyoupin/latest" path="/xiaomiyoupin/latest" />
```

File: lib/v2/xiaomiyoupin/maintainer.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    '/crowdfunding': ['bigfei'],
+    '/latest': ['bigfei'],
```
Comment (TonyRL) [inline lib/v2/xiaomiyoupin/maintainer.js:3]:
```suggestion
    '/latest': ['xyqfer', 'DIYgod', 'bigfei'],
```

File: lib/v2/xiaomiyoupin/crowdfunding.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const pageData = await got(`${base_url}/mtop/navi/venue/page?page_id=${pageid}&pdl=jianyu&sign=${sign}`);
+    const crowd_funding_floor = pageData.data.data.floors.find((floor) => floor.module_key === 'crowding');
+    const query_list = crowd_funding_floor.query_list;
+
+    // 2. fetchFloorDetailData
+    const floor_detail_data = await got.post(`${base_url}/mtop/navi/venue/batch?page_id=${pageid}&pdl=jianyu&sign=${sign}`, {
+        json: {
+            query_list,
+        },
+    });
```
Comment (TonyRL) [inline lib/v2/xiaomiyoupin/crowdfunding.js:22]: I think you can reuse the code in `latest.js` but a `module_key` filter of `crowd_funding`

File: lib/v2/xiaomiyoupin/utils.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const parseModule = (floors, module_key) => floors.filter((floor) => floor.module_key === module_key)[0];
+
+const parseFloorItem = (floor) =>
+    floor.data.items.map((item) => {
+        const i = item.item;
+        return {
+            title: i.name,
+            link: i.jump_url,
+            guid: `xiaomiyoupin:${i.gid}`,
+            description: art(path.join(__dirname, 'templates/goods.art'), i),
```
[the remark this answers is not in the document]
Comment (bigfei) [reply lib/v2/xiaomiyoupin/utils.js:13]: for `crowd_funding` items, there is `i.start`, but no explicit date is provided for `product_hot` items.

[the remark this answers is not in the document]
Comment (TonyRL) [reply lib/v2/xiaomiyoupin/utils.js:13]: `i.start ? parseDate(i.start, X) : undefined`

**Pull request merged**